File: Santander/app/src/main/java/teste/santander/com/santander/MainConfigurator.java
package teste.santander.com.santander;

import java.lang.ref.WeakReference;

public enum MainConfigurator {
    INSTANCE;
    public void configure(MainActivity activity){

        MainRouter router = new MainRouter();
        router.activity = new WeakReference<>(activity);

        MainPresenter presenter = new MainPresenter();
        presenter.output = new WeakReference<MainActivityInput>(activity);

        MainInteractor interactor = new MainInteractor();
        interactor.output = presenter;
        interactor.activity = new WeakReference<>(activity);

        if (activity.output == null){
            activity.output = interactor;
        }

    }
}

File: app/src/main/java/com/accenture/primo/bankapp/ui/main/MainConfigurator.kt
package com.accenture.primo.bankapp.ui.main

enum class MainConfigurator {
    INSTANCE;

    fun configure(activity: MainActivity) {
        val router = MainRouter(activity)
        val presenter = MainPresenter(activity)
        val interactor = MainInteractor(presenter)

        if (activity.interactor == null)
            activity.interactor = interactor

        if (activity.router == null)
            activity.router = router
    }
}
